### PR TITLE
tests for eda roles

### PR DESCRIPTION
**Pull request by appuk**
https://issues.redhat.com/browse/AAP-21630

- [x] unskip other role tests
- [x] add tests for custom roles

### CODE REVIEW

**Comment by appuk** [comment]
@akus062381 @vishsanghishetty @vidyanambiar I didn't create custom command for creating/ deleting the custom roles as it is only going to be used in 3 tests. let me know if you still want me to write a custom command for it

**Comment by appuk** [comment]
@vidyanambiar not sure why lint is failing, can you help me here? Thanks!!

**Comment by vidyanambiar** [comment]
@appuk Thanks for adding these tests!

The linter error is caused by references to old role interfaces `EdaRole` and `RoleDetail` in the cypress commands/tests. If you replace these references with the interface `EdaRbacRole`, the errors should go away.

File: cypress/e2e/eda/Roles/custom-roles-crud.cy.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    cy.get('h1').should('contain', 'Roles');
+    cy.clickTableRow(name, true);
+    cy.clickButton(/^Edit role$/);
+    cy.verifyPageTitle(`Edit Role`);
+    cy.get('[data-cy="description"]').type('this custom role has been changed');
+    cy.clickButton(/^Save role$/);
+    cy.hasDetail('Description', 'this custom role has been changed');
+  });
+
+  it('can delete a custom role', () => {
```
Comment (akus062381) [inline cypress/e2e/eda/Roles/custom-roles-crud.cy.ts:65]: Hi Apurva! I ran the tests on this spec, and saw the following error with this test:
image

File: cypress/e2e/eda/Roles/eda-role-details.cy.ts [hunk 2]
```diff
@@ -1,31 +1,24 @@
-// describe('Eda Role Details', () => {
```
Comment (akus062381) [inline cypress/e2e/eda/Roles/eda-role-details.cy.ts:1]: This spec file looks good!

File: cypress/e2e/eda/Roles/eda-roles-list.cy.ts [hunk 3]
```diff
@@ -1,23 +1,23 @@
-/** The Roles endpoint is going to change with the EDA server's RBAC branch and these tests will need to be updated to reflect the new endpoint */
 //Tests a user's ability to perform certain actions on the Roles list in the EDA UI.
```
Comment (akus062381) [inline cypress/e2e/eda/Roles/eda-roles-list.cy.ts:1]: this spec file looks good!

**Review by akus062381 — CHANGES_REQUESTED** [review]

**Review by vishsanghishetty — APPROVED** [review]
nice work!

**Review by akus062381 — APPROVED** [review]
Looking good! good work

**Pull request merged**